(closes #279) Remove Arguments.dofs and generally improve code coverage

sergisiso opened the pull request.

arporter commented on src/psyclone/psyir/nodes/node.py:
```diff
@@ -57,6 +57,7 @@
+# no cover: start
 try:
     # pylint disable=import-outside-toplevel
     from termcolor import colored
```
Can't we monkeypatch the packages that appear to be available from `termcolor`? I had a google and found https://stackoverflow.com/questions/8658043/how-to-mock-an-import

arporter commented on src/psyclone/tests/gocean1p0_test.py:
```diff
@@ -115,6 +115,29 @@ def test_field(tmpdir, dist_mem):
 
 
+def test_invalid_field_accesses_for_iteration_space():
+    ''' Tests that we when retrieving the iteration_space_arg we validate
```
s/we when/when/

arporter commented on src/psyclone/tests/psyir/frontend/fparser2_program_handler_test.py:
```diff
@@ -47,6 +47,20 @@
 
 
+def test_empty_program_handler(parser):
+    '''Test that an empty program handler still returns a FileContainer
```
"the program handler still returns a FileContainer for an empty program" perhaps?

sergisiso replied:
Yes, that's better

arporter commented on src/psyclone/transformations.py:
```diff
@@ -2115,35 +2114,12 @@ def apply(self, node: Schedule, options: Optional[Dict[str, Any]] = {}):
         # extract async. Default to False.
         async_queue = options.get('async_queue', False)
 
-        # check
```
How come you've removed this? I see we still support the specification of an `async_queue`?

sergisiso replied:
I reverted this change as it was not the purpose of this PR (but I think we can probably delete this validation in the future)

arporter reviewed (changes requested):
Thanks very much for tackling this Sergi.
There are a couple of cases where I think we can add tests rather than no-cover pragmas. Also, you've removed some validation code from ACCEnterDataTrans and I don't understand why.

sergisiso commented on src/psyclone/gocean1p0.py:
```diff
@@ -1207,13 +1206,6 @@ def find_grid_access(self):
         # if the kernel requires a grid-property argument.
         return None
 
-    @property
```
Great!

arporter reviewed (changes requested):
Thanks Sergi. Apart from the need to put back (or discuss) the check for different asynch. queues in EnterData, this is pretty much ready to go now. I see the integration tests were all OK previously so won't re-run them now.

sergisiso commented:
I reverted the controversial change and removed #3250 from the list of issues that the PR will close since finally it doesn't reach 100% coverage and I still want to look to the remaining lines at some point.

Ready for another look @arporter

arporter reviewed (approved):
All looks good now, thanks Sergi.
As master has now moved on, I'll re-run the IT and then proceed to merge.
